query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/insert/InsertInto.java

```diff
@@ -31,4 +37,29 @@
   /** Makes this statement an INSERT JSON with a bind marker, as in {@code INSERT JSON ?}. */
   @NonNull
   JsonInsert json(@NonNull BindMarker bindMarker);
+
+  /**
+   * Makes this statement an INSERT JSON with a custom type mapping, as in {@code INSERT JSON ?}.
+   *
+   * <p>This is an alternative to {@link #json(String)} for custom type mappings. The provided
+   * registry should contain a codec that can format the value. Typically, this will be your
+   * session's registry, which is accessible via {@code session.getContext().getCodecRegistry()}.
+   *
+   * @throws CodecNotFoundException if {@code codecRegistry} does not contain any codec that can
+   *     handle {@code value}.
+   * @see DriverContext#getCodecRegistry()
+   * @see DefaultInsert#json(Object, CodecRegistry)
+   */
+  @NonNull
+  JsonInsert json(@Nullable Object value, @NonNull CodecRegistry codecRegistry);
+
+  /**
+   * Makes this statement an INSERT JSON with a custom type mapping, as in {@code INSERT JSON ?}.
+   * The value will be turned into a string with {@link TypeCodec#format(Object)}, and inlined in
+   * the query.
+   *
+   * @see DefaultInsert#json(T, TypeCodec)
+   */
+  @NonNull
+  <T> JsonInsert json(@Nullable T value, @Nullable TypeCodec<T> codec);
```

Comment:
yes, you right. Changed

Reply:
After checking again the `codec` can be null because it proxy that into the `literal`. And `literal` is taking `codec` that is `@Nullable`:
```
literal(@Nullable T value, @Nullable TypeCodec<T> codec)
```
This is a case when the `value` that is passed to the `json` is `null`. Then the logic is:
```
value, (value == null) ? null : codecRegistry.codecFor(value)
```